int64 size
---
int64 blur_window_size
